_posts/en/newsletters/2019-05-28-newsletter.md

```diff
@@ -0,0 +1,597 @@
+amount of security as the confirmed setup transaction.  At any time, any
+of the receivers who wants to spend their money further can broadcast
+the distribution transaction and wait for it to confirm (perhaps helping
+to pay its fee using child-pays-for-parent fee bumping).
+
+This technique allows spenders and receivers to lock in a set of
+payments during high fees and then only distribute the actual payments when
+fees are lower.  According to Bitcoin Core fee estimates at the time of
+writing, anyone patient enough to wait a week for a transaction to
+confirm (like the distribution transaction above) can save 99% on fees.
+Let's look at the example above in that context.  To make later
+comparisons to Taproot more fair, we'll assume some form of key and
+signature aggregation is being used, such as [MuSig][] or (in theory)
+multiparty ECDSA (see [Newsletter #18][]).
+
+| | Individual Payments | Batched Payment | Commit now, distribute later |
+|-|-|-|-|
+| Immediate (high fee) transactions | 10x141 vbytes | 1x411 vbytes | 1x141 vbytes |
+| Cost at 0.00142112 BTC/KvB | 0.00204641 | 0.00058409 | 0.0002003 |
+| Delayed (low fee) transactions | --- | --- | 1x381 vbytes |
+| Cost at 0.00001014 BTC/KvB | --- | --- | 0.00000386 |
+| Total vbytes | 1,410 | 411 | 522 |
+| Total cost | 0.00204641 | 0.00058409 | 0.00020416 |
+| **Savings compared to previous column** | --- | **71%** | **65%** |
+
+We see that this type of trustlessly delayed payment can save 65% over
+payment batching and 90% over sending separate payments.  Note that the
+savings could be even larger during periods of greater fee
+stratification or with more than ten receivers.
+
+### CheckOutputsHashVerify
+
+The proposed soft fork would add a new opcode,
+`OP_CHECKOUTPUTSHASHVERIFY` (abbreviated by its author as `OP_COSHV`
+with an extra *S*).  This opcode and a hash digest could be included in
+tapleaf scripts, allowing it to be one of the conditions in a Taproot
+address.  When that address was spent, if COSHV was executed, the
+spending transaction would only be valid if the hash digest of its
+outputs matched the hash digest read from the script by COSHV.
+
+Comparing this to our example above, Alice would again ask each of the
+participants for a public key (such as a Taproot
+address[^taproot-pubkeys]).  Similar to before, she'd create 10 outputs
+which each paid one of the receivers---but she wouldn't need to form
+this into a specific distribution transaction.  Instead, she'd just hash
+the ten outputs together and use the resultant digest to create a
+tapleaf script containing COSHV.  That would be the only tapleaf in this
+Taproot commitment.  Alice could also use the participants' public keys
+to form the taproot internal key to allow them to cooperatively spend
+the money without revealing the Taproot script path.
+
+![FIXME](/img/posts/2019-05-tx-tree-2.png)
+
+Alice could then give each of the receivers a copy of all ten outputs to
```

Comment:
Alice has to give all the receivers a copy of all the outputs, otherwise they can't claim their specific output.